### PR TITLE
[AMD][CI] Fix Language Models Test (Extended Generation) failures

### PR DESCRIPTION
**Pull request by okorzh-amd**
## Purpose

Two unrelated failures in
tests/models/language/generation/test_common.py::test_models:

- openbmb/MiniCPM4.1-8B: its remote modeling_minicpm.py imports `is_torch_fx_available` from `transformers.utils.import_utils`, which was removed in Transformers v5, so the HF reference run fails to load with an ImportError. Cap the HF run at transformers 4.57 via max_transformers_version so the comparison is skipped on v5 (vLLM's native impl is unaffected).

- TitanML/tiny-mixtral (AITER): the untrained MoE has near-uniform router logits, so the AITER fused-MoE bf16 rounding flips expert selection and the final token diverges from HF even with AITER rms_norm already disabled. Also route the fused MoE through the native kernel for this model.

## Test Plan
Run failed tests

## Test Result
=========================== short test summary info ============================
PASSED tests/models/language/generation/test_common.py::test_models[True-True-5-32-TitanML/tiny-mixtral]
PASSED tests/models/language/generation/test_common.py::test_models[True-False-5-32-TitanML/tiny-mixtral]
PASSED tests/models/language/generation/test_common.py::test_models[False-True-5-32-TitanML/tiny-mixtral]
PASSED tests/models/language/generation/test_common.py::test_models[False-False-5-32-TitanML/tiny-mixtral]
SKIPPED [4] tests/models/registry.py:173: `transformers==5.11.0` installed, but `transformers<=4.57` is required to run this model. Reason(hf): Remote modeling_minicpm.py imports `is_torch_fx_available` from `transformers.utils.import_utils`, which was removed in Transformers v5.
===== 4 passed, 4 skipped, 72 deselected, 46 warnings in 197.56s (0:03:17) =====
EXIT=0

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
</details>

### CODE REVIEW

File: tests/models/language/generation/test_common.py [hunk 1]
```diff
@@ -130,24 +134,18 @@ def test_models(
         monkeypatch.setenv("VLLM_ROCM_USE_AITER", "1")
         if model == "TitanML/tiny-mixtral":
             # Untrained model: near-uniform logits make argmax sensitive to
-            # AITER's bfloat16 rounding error in plain rms_norm.
+            # AITER's bfloat16 rounding error. Route the plain rms_norm and the
+            # fused MoE (whose near-uniform router logits flip expert selection
+            # under ~1 ULP drift) through the native kernels for this model.
             monkeypatch.setenv("VLLM_ROCM_USE_AITER_RMSNORM", "0")
+            monkeypatch.setenv("VLLM_ROCM_USE_AITER_MOE", "0")
```
Comment (AndreasKaratzas) [inline tests/models/language/generation/test_common.py:138]: Please link an issue on AITER repo that includes minimal repro as well.

File: tests/models/registry.py [hunk 2]
```diff
@@ -405,7 +419,16 @@ def check_available_online(
         "openbmb/MiniCPM3-4B", trust_remote_code=True
     ),
     "MiniCPM4ForCausalLM": _HfExamplesInfo(
-        "openbmb/MiniCPM4.1-8B", trust_remote_code=True
+        "openbmb/MiniCPM4.1-8B",
```
Comment (AndreasKaratzas) [inline tests/models/registry.py:414]: cc @hmellor PTAL

**Comment by DarkLight1337** [comment]
cc @tc-mb

**Comment by tc-mb** [comment]
> cc @tc-mb

Received, I'll check it soon.

**Comment by tc-mb** [comment]
Hi @okorzh-amd, thanks for fixing these two CI issues! Both changes look correct to me as they stand — they'll get CI green.

That said, I think the MiniCPM4.1-8B fix could go a step further. `max_transformers_version="4.57"` skips the test, but doesn't actually let the model work on transformers v5. The root cause is that `is_torch_fx_available` was removed in v5, but the remote `modeling_minicpm.py` only uses it in one place (to wrap the attention mask with `torch.fx.wrap` for FX tracing — doesn't affect inference at all). Since `torch.fx` is always available in PyTorch 2.0+, we could just shim it back in `vllm/transformers_utils/config.py` before loading remote code. Then MiniCPM4 would actually work on v5 rather than being capped out.

Would you be interested in taking that approach, or would you prefer I follow up with a separate PR? Happy either way.

(cc @DarkLight1337)

**Comment by tc-mb** [comment]
I opened a PR against your branch with the alternative shim approach, feel free to take a look: https://github.com/okorzh-amd/vllm/pull/2. Happy either way!

**Comment by okorzh-amd** [comment]
> Hi @okorzh-amd, thanks for fixing these two CI issues! Both changes look correct to me as they stand — they'll get CI green.
> 
> That said, I think the MiniCPM4.1-8B fix could go a step further. `max_transformers_version="4.57"` skips the test, but doesn't actually let the model work on transformers v5. The root cause is that `is_torch_fx_available` was removed in v5, but the remote `modeling_minicpm.py` only uses it in one place (to wrap the attention mask with `torch.fx.wrap` for FX tracing — doesn't affect inference at all). Since `torch.fx` is always available in PyTorch 2.0+, we could just shim it back in `vllm/transformers_utils/config.py` before loading remote code. Then MiniCPM4 would actually work on v5 rather than being capped out.
> 
> Would you be interested in taking that approach, or would you prefer I follow up with a separate PR? Happy either way.
> 
> (cc @DarkLight1337)

Thanks @tc-mb ! You're right that the shim would work and it's low-risk since torch.fx is always present.

One clarification though: vLLM inference of MiniCPM4 already works on transformers v5 as we use vLLM's native MiniCPM4ForCausalLM, not the remote modeling_minicpm.py. That remote file is only pulled in by the HF reference run in this parity test, so the cap just skips the HF-vs-vLLM comparison rather than blocking real usage.

My concern is that special-casing one model's stale remote code inside transformers_utils/config.py adds a model-specific patch to core, which is inconsistent with how we already handle every other model with broken remote code (the max_transformers_version + transformers_version_reason={"hf": ...} pattern: ChatGLM, Qwen-VL, Molmo, Phi-3-vision, Skywork, etc.). Since the offending import (is_torch_fx_available, used only for FX tracing, line 359) doesn't affect inference at all, the cleanest fix is on the model side.

@tc-mb would you be able to update the remote modeling_minicpm.py (drop the removed import / guard it)? Once that lands we can lift the cap. I'd prefer to keep this PR as the interim fix rather than add a shim to core but no objection if you'd still like to explore the shim in a separate PR.

**Comment by tc-mb** [comment]
Thanks for the thoughtful response @okorzh-amd! Fair point on consistency — the max_transformers_version pattern is the established way to handle stale remote code, and special-casing one model in core doesn't make sense. I've closed the alternative PR.

I'm happy with this PR as the interim fix — let's land it as-is. I'll follow up by opening a PR on openbmb/MiniCPM4.1-8B to fix the import in the remote modeling file, then we can come back and lift the version cap.

(cc @DarkLight1337)

**Review by DarkLight1337 — APPROVED** [review]

**Comment by DarkLight1337** [comment]
Please resolve the merge conflict

File: tests/models/language/generation/test_common.py (hunk 1 above)
Comment (okorzh-amd) [reply tests/models/language/generation/test_common.py:138]: here https://github.com/ROCm/aiter/issues/3806

**Comment by okorzh-amd** [comment]
> Please resolve the merge conflict

https://github.com/vllm-project/vllm/pull/41532 this one merged the change for MiniCPM4  . So this PR is now aiter issue only

File: tests/models/language/generation/test_common.py (hunk 1 above)
Comment (AndreasKaratzas) [reply tests/models/language/generation/test_common.py:138]: I meant in the comment 😅

Comment (okorzh-amd) [reply tests/models/language/generation/test_common.py:138]: dane
